tests/e2e/multichain/burn_test.go

```diff
@@ -0,0 +1,225 @@
+	defaultDepositAmount = sdk.NewCoin(denom, sdkmath.NewInt(10_000_000))
+	lessDepositAmount    = sdk.NewCoin(denom, sdkmath.NewInt(100_000))
+	govAddress           string
+	validatorKeyName     = "validator"
+)
+
+type testResult struct {
+	proposal            *govv1types.Proposal
+	status              govv1types.ProposalStatus
+	diffSupplyBalance   sdkmath.Int
+	diffProposerBalance sdkmath.Int
+}
+
+func TestMsgBurn(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	ctx := context.Background()
+	chain, _ := multichain.StartXplaChain(t, ctx, multichain.LocalImage)
+
+	xplaDefaultBalance, _ := sdkmath.NewIntFromString("10_000_000_000_000_000_000")
+	xplaUsers := interchaintest.GetAndFundTestUsers(t, ctx, "default", xplaDefaultBalance, chain)
+	user := xplaUsers[0]
+
+	govAddress, _ = chain.AuthQueryModuleAddress(ctx, govtypes.ModuleName)
+
+	tests := []struct {
+		title         string
+		depositAmount sdk.Coin
+		voteOpt       govv1types.VoteOption
+		expected      testResult
+	}{
+		{
+			"MsgBurn proposal, Vote YES",
+			defaultDepositAmount,
+			govv1types.VoteOption_VOTE_OPTION_YES,
+			testResult{
+				status:              govv1types.ProposalStatus_PROPOSAL_STATUS_PASSED,
+				diffSupplyBalance:   defaultBurnAmount.Amount,
+				diffProposerBalance: defaultDepositAmount.Amount,
+			},
+		},
+		{
+			"MsgBurn proposal, Vote NO",
+			defaultDepositAmount,
+			govv1types.VoteOption_VOTE_OPTION_NO,
+			testResult{
+				status:              govv1types.ProposalStatus_PROPOSAL_STATUS_REJECTED,
+				diffSupplyBalance:   sdkmath.ZeroInt(),
+				diffProposerBalance: defaultDepositAmount.Amount.Add(defaultBurnAmount.Amount),
+			},
+		},
+		{
+			"MsgBurn proposal, Vote ABSTAIN",
+			defaultDepositAmount,
+			govv1types.VoteOption_VOTE_OPTION_ABSTAIN,
+			testResult{
+				status:              govv1types.ProposalStatus_PROPOSAL_STATUS_REJECTED,
+				diffSupplyBalance:   sdkmath.ZeroInt(),
+				diffProposerBalance: defaultDepositAmount.Amount.Add(defaultBurnAmount.Amount),
+			},
+		},
+		{
+			"MsgBurn proposal, Vote UNSPECIFIED",
+			defaultDepositAmount,
+			govv1types.VoteOption_VOTE_OPTION_UNSPECIFIED,
+			testResult{
+				status:              govv1types.ProposalStatus_PROPOSAL_STATUS_REJECTED,
+				diffSupplyBalance:   sdkmath.ZeroInt(),
+				diffProposerBalance: defaultDepositAmount.Amount.Add(defaultBurnAmount.Amount),
+			},
+		},
+		{
+			"MsgBurn proposal, Vote NO_WITH_VETO",
+			defaultDepositAmount,
+			govv1types.VoteOption_VOTE_OPTION_NO_WITH_VETO,
+			testResult{
+				status:              govv1types.ProposalStatus_PROPOSAL_STATUS_REJECTED,
+				diffSupplyBalance:   defaultDepositAmount.Amount,
+				diffProposerBalance: defaultBurnAmount.Amount,
+			},
+		},
+		{
+			"MsgBurn proposal, Vote UNSPECIFIED, less deposit",
+			lessDepositAmount,
+			govv1types.VoteOption_VOTE_OPTION_UNSPECIFIED,
+			testResult{
+				diffSupplyBalance:   sdkmath.ZeroInt(),
+				diffProposerBalance: lessDepositAmount.Amount.Add(defaultBurnAmount.Amount),
```

Comment:
AFIK the deposits are burned if they are rejected due to 'less deposit', so I believe this part should be modified.

Reply:
```
"burn_vote_quorum": false,
"burn_proposal_deposit_prevote": false,
"burn_vote_veto": true,
"min_deposit_ratio": "0.010000000000000000"
```

Currently, it seems that there are no cases that burn except for veto, but do you want to cover all such cases?

Reply:
I wasn't aware it depends on param and the default value was false. However, if the result can change based on the params value, then either fix the params value in test or get the params and make the behavior dependent on it, I think.

Reply:
But I think we can just leave it for now. ;)